packages/frontend/src/components/RichTextEditor/StepVariablePlugin.ts

```diff
@@ -0,0 +1,104 @@
+import { mergeAttributes, Node } from '@tiptap/core'
+import { PluginKey } from '@tiptap/pm/state'
+import { ReactNodeViewRenderer } from '@tiptap/react'
+
+import { VariableBadge } from './VariableBadge'
+
+export type VariableOptions = {
+  HTMLAttributes: Record<string, any>
+}
+const key = 'variable'
+export const VariablePluginKey = new PluginKey(key)
+
+export const StepVariable = Node.create<VariableOptions>({
+  name: key,
+  addOptions() {
+    return {
+      HTMLAttributes: {},
+    }
+  },
+  group: 'inline',
+  inline: true,
+  selectable: false,
+  atom: true,
+  addAttributes() {
+    // attributes equivalent of
+    // id: step name
+    // label: step label
+    // value: step test value
+    // these are only used for NodeView purpose
+    // we technically don't need to render them into the final HTML value, however
```

Comment:
This is OK but it looks like we're relying on postman to strip these `data-xxx` attributes out for us when we send out the email? Can we strip these ourselves before calling postman? Reasons:
1. Reduces user confusion - currently, after clicking "Test Step", user will see these attributes in the `Data` field and might go wth?
2. Remove dependency on external service santitizing our html. (as pointed out by @liangyuanruo )

Reply:
As discussed we don't need to strip